dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcTrackedEntityEventsAnalyticsTableManager.java

```diff
@@ -88,30 +88,6 @@
 @Component("org.hisp.dhis.analytics.TrackedEntityEventsAnalyticsTableManager")
 public class JdbcTrackedEntityEventsAnalyticsTableManager extends AbstractJdbcTableManager {
 
-  private static final String EVENT_DATA_VALUE_REBUILDER =
-      """
-      (select json_object_agg(l2.keys, l2.datavalue) as value
-      from (
-          select l1.uid,
-          l1.keys,
-          json_strip_nulls(json_build_object(
-          'value', l1.eventdatavalues -> l1.keys ->> 'value',
-          'created', l1.eventdatavalues -> l1.keys ->> 'created',
-          'storedBy', l1.eventdatavalues -> l1.keys ->> 'storedBy',
-          'lastUpdated', l1.eventdatavalues -> l1.keys ->> 'lastUpdated',
-          'providedElsewhere', l1.eventdatavalues -> l1.keys -> 'providedElsewhere',
-          'value_name', (select ou.name
-              from organisationunit ou
-              where ou.uid = l1.eventdatavalues -> l1.keys ->> 'value'),
-          'value_code', (select ou.code
-              from organisationunit ou
-              where ou.uid = l1.eventdatavalues -> l1.keys ->> 'value'))) as datavalue
-          from (select inner_evt.*, jsonb_object_keys(inner_evt.eventdatavalues) keys
-          from event inner_evt) as l1) as l2
-      where l2.uid = ev.uid
-      group by l2.uid)::jsonb
-      """;
```

Comment:
I agree we should not remove existing features or introducing regression.
One possibility I can think of, is to add a condition based on the underlying analtyics database. Something like:

```
if (sqlBuilder.getDatabase().equals(POSTGRES)) {
  // Add removed query
}
```

Reply:
@gnespolino you think you could add Javadoc/explanation of what the logic is for this expression? I am trying to understand what is going on but it is not trivial.

Reply:
We can keep this SQL JSON expression for PG, and simplify for Doris/CH for now.

Reply:
Hey @larshelge,

In tracked entity analytics, unlike enrollment and event analytics, we decided to keep eventdatavalues as a JSONB for several reasons—primarily because it's multi-program, meaning a data element could exist in one program but not in another.

In event and enrollment analytics, when a data value (with UID = X) is of type ORG_UNIT, we add two extra fields: X_NAME and X_CODE. Specifically, we take the JSON, flatten it, add the entries value_name and value_code, and then reconstruct the JSONB. This process happens "online" during the analytics export.

Example:
Given this event.eventdatavalues structure:

```javascript
{
  ...
  "dataelementUid": {
    ...
    "value": "orgunitUid"
    ...
  }
  ...
}
```

It would be transformed into:

```javascript
{
  ...
  "dataelementUid": {
    ...
    "value": "orgunitUid",
    "value_name": "orgunitName",
    "value_code": "orgunitCode"
    ...
  }
  ...
}
```

If we merge this PR, while event.eventdatavalues would still be structurally compatible with the analytics version, it will lack the essential value_name and value_code fields when it contains a data element of type ORG_UNIT. These fields are crucial for maintaining functionality and ensuring org unit details are properly represented.

Reply:
No worries, I will add a comment, I am about to push some code that will preserve the original functionality.

Reply:
Thanks for the explanation @gnespolino !